invenio_rdm_records/records/dumpers/record_status.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""DataCite-based data model for Invenio."""
+
+from invenio_records.dictutils import dict_set
+from invenio_records.dumpers import ElasticsearchDumperExt
+
+
+class RecordStatusDumperExt(ElasticsearchDumperExt):
```

Comment:
I think this has to be a system field rather than only something available in the index. E.g. I would assume we would expose the status in the REST API JSON, so that we can e.g. display the right icon.